error.go

```diff
@@ -0,0 +1,69 @@
+package semver
+
+import (
+	"bytes"
+	"fmt"
+)
+
+var rangeErrs = [...]string{
+	"%s is less than the minimum of %s",
+	"%s is less than or equal to the minimum of %s",
+	"%s is greater than the maximum of %s",
+	"%s is greater than or equal to the maximum of %s",
+	"%s is specifically disallowed in %s",
+}
+
+const (
+	rerrLT = iota
+	rerrLTE
+	rerrGT
+	rerrGTE
+	rerrNE
+)
+
+type MatchFailure interface {
+	error
+	Pair() (v *Version, c Constraint)
+}
+
+type rangeMatchFailure struct {
+	v   *Version
+	rc  rangeConstraint
+	typ int8
+}
```

Comment:
Not a lot of thinking on this one, actually :) I did consider making them public, and I'd be fine to do that still. Just, can't expose the fields; `rangeConstraint`s are and should remain internal, i think